presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -447,15 +452,24 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                 }
             }
 
-            // See if we can push down any outer or join predicates to the inner side
-            for (Expression conjunct : EqualityInference.nonInferrableConjuncts(and(outerEffectivePredicate, joinPredicate))) {
+            // See if we can push down any outer predicates to the inner side
+            for (Expression conjunct : EqualityInference.nonInferrableConjuncts(outerEffectivePredicate)) {
                 Expression rewritten = potentialNullSymbolInference.rewriteExpression(conjunct, not(in(outerSymbols)));
                 if (rewritten != null) {
                     innerPushdownConjuncts.add(rewritten);
                 }
             }
 
-            // TODO: consider adding join predicate optimizations to outer joins
+            // See if we can push down any join predicates to the inner side
+            for (Expression conjunct : extractConjuncts(joinPredicate)) {
+                Expression rewritten = innerInference.rewriteExpression(conjunct, not(in(outerSymbols)));
```

Comment:
AH! Yea that will happen if you don't do:

EqualityInference.nonInferrableConjuncts(joinPredicate)
try that instead of:
extractConjuncts(joinPredicate)

you'll see all the other rewrite code paths do that instead